samples/nfc/tag_reader/src/main.c

```diff
@@ -350,12 +402,61 @@ static const struct st25r3911b_nfca_cb cb = {
 	.tag_sleep = tag_sleep
 };
 
+static void t4t_isodep_selected(const struct nfc_t4t_isodep_tag *t4t_tag)
+{
+	printk("NFC T4T selected.\n");
+
+	if ((t4t_tag->lp_divisor != 0) &&
+	    (t4t_tag->lp_divisor != t4t_tag->pl_divisor)) {
+		printk("Unsupported bitrate divisor by Reader/Writer.\n");
+	}
+
+	st25r3911b_nfca_tag_sleep();
+
+	k_delayed_work_submit(&transmit_work, TRANSMIT_DELAY);
+}
+
+static void t4t_isodep_error(int err)
+{
+	printk("ISO-DEP Protocol error %d\n.", err);
+
+	nfc_tag_detect(false);
+}
+
+static void t4t_isodep_data_send(u8_t *data, size_t data_len, u32_t ftd)
+{
+	int err;
+
+	tx_buf.data = data;
+	tx_buf.len = data_len;
+
+	err = st25r3911b_nfca_transfer_with_crc(&tx_buf, &rx_buf, ftd);
+	if (err) {
+		printk("NFC T4T ISO-DEP transfer error: %d\n.", err);
+	}
+}
+
+static const struct nfc_t4t_isodep_cb t4t_isodep_cb = {
+	.selected = t4t_isodep_selected,
+	.error = t4t_isodep_error,
+	.ready_to_send = t4t_isodep_data_send
+};
+
 void main(void)
 {
 	int err;
 
 	printk("NFC reader sample started\n");
 
+	err = nfc_t4t_isodep_init(tx_data, sizeof(tx_data),
+				  t4t.data, sizeof(t4t.data),
+				  &t4t_isodep_cb);
+	if (err) {
+		printk("NFC T4T ISO-DEP Protocol initialization failed err: %d\n",
+		       err);
+		return;
+	}
+
 	k_delayed_work_init(&transmit_work, transfer_handler);
```

Comment:
isodep init is done before work is initialized. I would move work init to the top

Reply:
Ok. Moved